File: srcJava/com/rti/perftest/IMessagingWriter.java
/*
 * (c) 2005-2017  Copyright, Real-Time Innovations, Inc. All rights reserved.
 * Subject to Eclipse Public License v1.0; see LICENSE.md for details.
 */

package com.rti.perftest;

import java.util.concurrent.TimeUnit;

// ===========================================================================

/**
 * Send messages.
 */
public interface IMessagingWriter {
    public void waitForReaders(int numReaders);

    public boolean send(TestMessage message, boolean isCftWildCardKey);
    public boolean waitForPingResponse();
    public boolean waitForPingResponse(long timeout, TimeUnit unit);
    public boolean notifyPingResponse();
    public long getPulledSampleCount();
    public void waitForAck(int sec, int nsec);
    public void flush();

}

// ===========================================================================

File: srcCs/Interface/IMessagingWriter.cs
/*
 * (c) 2005-2021 Copyright, Real-Time Innovations, Inc. All rights reserved.
 * Subject to Eclipse Public License v1.0; see LICENSE.md for details.
 */

using System;

namespace PerformanceTest
{
    public interface IMessagingWriter
    {
        void WaitForReaders(int numReaders);
        bool Send(TestMessage message, bool isCftWildCardKey);
        void Flush();
        bool NotifyPingResponse();
        bool WaitForPingResponse();
        bool WaitForPingResponse(TimeSpan timeout);
        long GetPulledSampleCount();
        void WaitForAck(TimeSpan timeSpan);
    }
}
